src/index.js

```
@@ -8,272 +8,358 @@ const _ = require("lodash");
+      ? JSON.parse(fs.readFileSync(testInputPath))
+      : "";
+    if (testInput) {
+      transformationTest.push({ versionId: trVersionId, testInput });
+    } else {
+      core.info(
+        `No test input provided. Testing ${transformationDict[trVersionId].name} with default payload`
+      );
+      transformationTest.push({ versionId: trVersionId });
     }
+  }
 
-    core.info(
-      `final transformation versions to be tested: 
-      ${JSON.stringify(transformationTest)}`
-    );
-    core.info(
-      `final library versions to be tested: ${JSON.stringify(librariesTest)}`
-    );
+  for (const versionId of Object.keys(libraryDict)) {
+    librariesTest.push({ versionId });
+  }
 
-    core.info("Running test...");
+  core.info(
+    `Final transformation versions to be tested:
+    ${JSON.stringify(transformationTest)}`
+  );
+  core.info(
+    `Final library versions to be tested: ${JSON.stringify(librariesTest)}`
+  );
+  return { transformationTest, librariesTest };
+}
 
-    let res = await testTransformationAndLibrary(
-      transformationTest,
-      librariesTest
+// Run the test suite.
+async function runTestSuite(transformationTest, librariesTest) {
+  core.info("Running test suite for transformations and libraries");
+
+  let res = await testTransformationAndLibrary(
+    transformationTest,
+    librariesTest
+  );
+
+  core.info(
+    colorize(
+      `\nTotal tests ${
+        res.data.result.successTestResults.length +
+        res.data.result.failedTestResults.length
+      }, ${res.data.result.successTestResults.length} passed and ${
+        res.data.result.failedTestResults.length
+      } failed\n`,
+      "yellow"
+    )
+  );
+  if (res.data.result.failedTestResults.length > 0) {
+    logFailedTests(res.data.result.failedTestResults);
+    throw new Error(
+      "Failures occured while running tests against input events"
     );
-    core.info(`Test api output: ${JSON.stringify(res.data)}`);
+  }
```

Let's pull this into a function called `logResults` to have better separation of concerns ?

---

Rather than loggingFailedTests, we log results which will include failed tests as well.

---

yeah so result object is also having a lot of extra data of transformation and libraries which could be the processing unnecessary slow, should we proceed with logging full result ?wdyt

---

We could just log the overview like successful and failed and then log failed. This is what we are doing right now, asking to move into a separate function.